test/models/secret_storage_test.rb

```diff
@@ -122,37 +183,46 @@
           SecretStorage::HashicorpVault.convert_path('foopy%2Fthecat', :notvalid)
         end
       end
+
+      it "recusivly translates keys" do
+        tree = ['this/is/still/a/branch/']
+        assert SecretStorage::HashicorpVault.keys_recursive(tree)
```

Comment:
also keys_recursive is private ? -> no direct testing needed ... test via public apis